src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -909,8 +913,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           checkParamsConvertible(tree, tree.tpe)
 
           // scala/bug#7187 eta-expansion of zero-arg method value is deprecated, switch order of (4.3) and (4.2) in 2.13
-          def isExplicitEtaExpansion = original match {
-            case Typed(_, Function(Nil, EmptyTree)) => true // tree shape for `f _`
+          def isExplicitEtaExpansion = original0 match {
+            case MethodValue(_) => true // tree shape for `f _`
             case _ => false
           }
```

Comment:
We could move this right up to the first use of `MethodValue.unapply`

```scala
val isExplicitEtaExpansion = original0 != original
```

Reply:
Yes.